source/appModules/code.py

```diff
@@ -21,11 +24,95 @@ class VSCodeDocument(Document):
 	_get_treeInterceptorClass = NVDAObject._get_treeInterceptorClass
 
 
+DIGIT_EXPR = re.compile(r"\d+")
+
+
 class AppModule(appModuleHandler.AppModule):
+	def __init__(self, *args, **kwargs):
+		super().__init__(*args, **kwargs)
+		self._status = None
+
+	@staticmethod
+	def _search_for_statusbar(root: NVDAObject) -> NVDAObject | None:
+		seen = set()
+		t = deque((root,))
+		while t:
+			obj = t.popleft()
+			if obj in seen:
+				continue
+			seen.add(obj)
+			try:
+				if obj.role == controlTypes.Role.STATUSBAR:
+					return obj
+				# IA2 ID often contains "statusbar"
+				ia2id = obj.IA2Attributes.get("id")
+				if ia2id and "statusbar" in ia2id.casefold():
+					return obj
+				children = obj.children
+			except Exception:
+				children = ()
+			t.extend(children)
+		return None
+
+	@staticmethod
+	def _looks_like_line_col(text: str) -> bool:
+		"""
+		Detect two integers separated by something that is NOT a dot,
+		to avoid version-like strings.
+		"""
+		it = DIGIT_EXPR.finditer(text)
+		first = next(it, None)
+		if first is None:
+			return False
+		second = next(it, None)
+		if second is None:
+			return False
+		between = text[first.end() : second.start()]
+		if "." in between:
+			return False
+		if not between.strip():
+			# Only whitespace or an empty string found,
+			# not the line/column number
+			return False
+		return True
+
+	def _get_statusBar(self) -> NVDAObject:
+		cached = self._status
+		if cached:
+			return cached
+
+		# Fallback: search the current foreground window tree for a STATUSBAR.
+		try:
+			foreground = api.getForegroundObject()
+		except Exception:
+			foreground = None
```

Comment:
I don't think should ever cause an exception

```suggestion
		foreground = api.getForegroundObject()
```